state/allwatcher.go

```diff
@@ -791,6 +798,16 @@ func (s *backingSettings) updated(st *State, store *multiwatcherStore, id string
 	case nil:
 		// The parent info doesn't exist. Ignore the status until it does.
 		return nil
+	case *multiwatcher.ModelInfo:
+		// We need to construct a model config so that coercion
+		// of raw settings values occurs.
+		cfg, err := config.New(config.UseDefaults, s.Settings)
```

Comment:
I think you should be using NoDefaults here, like in State.ModelConfig. Config should be complete when it's stored in state.